api/napi/src/interpreter/diagnostic.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use napi::bindgen_prelude::{FromNapiValue, ToNapiValue};
+use slint_interpreter::{Diagnostic, DiagnosticLevel};
+
+#[napi(js_name = "DiagnosticLevel")]
+pub enum JsDiagnosticLevel {
+    Error,
+    Warning,
+}
+
+impl From<DiagnosticLevel> for JsDiagnosticLevel {
+    fn from(diagnostic_level: DiagnosticLevel) -> Self {
+        match diagnostic_level {
+            DiagnosticLevel::Warning => JsDiagnosticLevel::Warning,
+            _ => JsDiagnosticLevel::Error,
+        }
+    }
+}
+
+#[napi(js_name = "Diagnostic")]
+pub struct JsDiagnostic {
+    internal: Diagnostic,
+}
+
+#[napi]
+impl JsDiagnostic {
+    #[napi]
+    pub fn level(&self) -> JsDiagnosticLevel {
+        self.internal.level().into()
+    }
+
+    #[napi]
+    pub fn message(&self) -> String {
+        self.internal.message().into()
+    }
+
+    #[napi]
+    pub fn line_column(&self) -> Vec<i32> {
+        let (start, column) = self.internal.line_column();
+        vec![start as i32, column as i32]
+    }
```

Comment:
I'm curious, is there a specific JavaScript API this mimics?

I'm thinking the idiomatic JavaScript API here would be to return an object that has a `lineNumber` and a `columNumber` property.